pkgs/top-level/all-packages.nix

```diff
@@ -9515,7 +9515,7 @@ with pkgs;
 
   nanoemoji = python3Packages.callPackage ../tools/misc/nanoemoji { };
 
-  nagelfar = callPackage ../development/tools/nagelfar { };
+  nagelfar = tclPackages.callPackage ../development/tools/nagelfar { };
```

Comment:
There are two ways to migrate top-level packages that use tcl packages. This is the first. The second is to keep the regular `callPackage` and get the tcl dependency from the `tclPackages` set, like this:

```diff
--- a/pkgs/applications/version-management/fossil/default.nix
+++ b/pkgs/applications/version-management/fossil/default.nix
@@ -12,7 +12,7 @@
 , sqlite
 , ed
 , which
-, tcllib
+, tclPackages
 , withJson ? true
 }:
 
@@ -28,7 +28,7 @@ stdenv.mkDerivation (finalAttrs: {
   # required for build time tool `./tools/translate.c`
   depsBuildBuild = [ buildPackages.stdenv.cc ];
 
-  nativeBuildInputs = [ installShellFiles tcl tcllib ];
+  nativeBuildInputs = [ installShellFiles tcl tclPackages.tcllib ];
 
   buildInputs = [ zlib openssl readline which ed ]
     ++ lib.optional stdenv.isDarwin libiconv
```

Using `tclPackages.callPackage` in `all-packages.nix` is incompatible with `by-name`, so I'll go with the other option. Other maintainers seem to agree https://github.com/NixOS/nixpkgs/pull/287169#discussion_r1483153767